qa/os/src/test/java/org/elasticsearch/packaging/util/docker/Docker.java

```diff
@@ -368,6 +371,24 @@ private static void executePrivilegeEscalatedShellCmd(String shellCmd, Path loca
         sh.run(command);
     }
 
+    private static void ensureImageIsPulled(String image) {
+        // Don't pull if the image already exists. This does also mean that we never refresh it, but that
+        // isn't an issue in CI.
+        if (sh.runIgnoreExitCode("docker image inspect -f '{{ .Id }}' " + image).isSuccess()) {
+            return;
+        }
+
+        Shell.Result result = null;
+        int i = 0;
+        while (i++ < 10) {
```

Comment:
I think we should probably do this fewer times (maybe 3) and put a slight delay in between each attempt.